diff -up Frodo-4.1b/Src/SID.cpp~ Frodo-4.1b/Src/SID.cpp
--- Frodo-4.1b/Src/SID.cpp~	2002-01-02 22:18:59.000000000 +0100
+++ Frodo-4.1b/Src/SID.cpp	2016-07-07 21:47:41.030508033 +0200
@@ -801,7 +801,7 @@ const uint8 DigitalRenderer::EGDRShift[2
 };
 
 const int16 DigitalRenderer::SampleTab[16] = {
-	0x8000, 0x9111, 0xa222, 0xb333, 0xc444, 0xd555, 0xe666, 0xf777,
+	(int16)0x8000, (int16)0x9111, (int16)0xa222, (int16)0xb333, (int16)0xc444, (int16)0xd555, (int16)0xe666, (int16)0xf777,
 	0x0888, 0x1999, 0x2aaa, 0x3bbb, 0x4ccc, 0x5ddd, 0x6eee, 0x7fff,
 };
 
